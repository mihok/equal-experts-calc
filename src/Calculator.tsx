import React, { useState, useRef } from "react";

const Calculator = () => {
  const [value, setValue] = useState("");
  const [action, setAction] = useState("");

  const performAction = () => {
    setValue(eval(value));
  };

  return (
    <div>
      <div className="output">{value}</div>
      <div>
        <button onClick={() => setValue(value + 7)} className="seven">
          7
        </button>
        <button onClick={() => setValue(value + 8)} className="eight">
          8
        </button>
        <button onClick={() => setValue(value + 9)} className="nine">
          9
        </button>

        <button onClick={() => setValue(value + 4)} className="four">
          4
        </button>
        <button onClick={() => setValue(value + 5)} className="five">
          5
        </button>
        <button onClick={() => setValue(value + 6)} className="six">
          6
        </button>

        <button onClick={() => setValue(value + 1)} className="one">
          1
        </button>
        <button onClick={() => setValue(value + 2)} className="two">
          2
        </button>
        <button onClick={() => setValue(value + 3)} className="three">
          3
        </button>

        <button onClick={() => setValue(value + 0)} className="zero">
          0
        </button>
        <button onClick={() => setValue(value + ".")} className="decimal">
          .
        </button>
      </div>
      <div>
        <button onClick={() => setValue(value + "/")} className="divided-by">
          รท
        </button>
        <button onClick={() => setValue(value + "*")} className="multiplied-by">
          ร
        </button>
        <button onClick={() => setValue(value + "-")} className="minus">
          -
        </button>
        <button onClick={() => setValue(value + "+")} className="plus">
          +
        </button>

        <button onClick={performAction} className="equals">
          =
        </button>
      </div>
    </div>
  );
};

export default Calculator;
