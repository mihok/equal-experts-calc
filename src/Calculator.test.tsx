import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import Calculator from "./Calculator";

describe("Calculator", () => {
  test("renders correctly", () => {
    const snapshot = shallow(<Calculator />);

    expect(toJson(snapshot)).toMatchSnapshot();
  });

  test("performs addition", () => {
    const calc = mount(<Calculator />);

    const output = calc.find(".output");

    calc.find("button.one").simulate("click");

    expect(output.text()).toEqual("1");

    calc.find("button.plus").simulate("click");

    expect(output.text()).toEqual("1+");

    calc.find("button.one").simulate("click");
    calc.find("button.equals").simulate("click");

    expect(output.text()).toEqual("2");
  });

  test("subtraction", () => {
    const calc = mount(<Calculator />);

    const output = calc.find(".output");

    calc.find("button.one").simulate("click");

    expect(output.text()).toEqual("1");

    calc.find("button.minus").simulate("click");

    expect(output.text()).toEqual("1-");

    calc.find("button.one").simulate("click");
    calc.find("button.equals").simulate("click");

    expect(output.text()).toEqual("0");
  });

  test("division", () => {
    const calc = mount(<Calculator />);

    const output = calc.find(".output");

    calc.find("button.one").simulate("click");
    calc.find("button.zero").simulate("click");

    expect(output.text()).toEqual("10");

    calc.find("button.divided-by").simulate("click");

    expect(output.text()).toEqual("10/");

    calc.find("button.five").simulate("click");
    calc.find("button.equals").simulate("click");

    expect(output.text()).toEqual("2");
  });

  test("multiplication", () => {
    const calc = mount(<Calculator />);

    console.log(toJson(calc));

    const output = calc.find(".output");

    calc.find("button.two").simulate("click");

    expect(output.text()).toEqual("2");

    calc.find("button.multiplied-by").simulate("click");

    expect(output.text()).toEqual("2*");

    calc.find("button.two").simulate("click");
    calc.find("button.equals").simulate("click");

    expect(output.text()).toEqual("4");
  });
});
