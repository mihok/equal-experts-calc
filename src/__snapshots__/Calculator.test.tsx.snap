// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div>
  <div
    className="output"
  />
  <div>
    <button
      className="seven"
      onClick={[Function]}
    >
      7
    </button>
    <button
      className="eight"
      onClick={[Function]}
    >
      8
    </button>
    <button
      className="nine"
      onClick={[Function]}
    >
      9
    </button>
    <button
      className="four"
      onClick={[Function]}
    >
      4
    </button>
    <button
      className="five"
      onClick={[Function]}
    >
      5
    </button>
    <button
      className="six"
      onClick={[Function]}
    >
      6
    </button>
    <button
      className="one"
      onClick={[Function]}
    >
      1
    </button>
    <button
      className="two"
      onClick={[Function]}
    >
      2
    </button>
    <button
      className="three"
      onClick={[Function]}
    >
      3
    </button>
    <button
      className="zero"
      onClick={[Function]}
    >
      0
    </button>
    <button
      className="decimal"
      onClick={[Function]}
    >
      .
    </button>
  </div>
  <div>
    <button
      className="divided-by"
      onClick={[Function]}
    >
      รท
    </button>
    <button
      className="multiplied-by"
      onClick={[Function]}
    >
      ร
    </button>
    <button
      className="minus"
      onClick={[Function]}
    >
      -
    </button>
    <button
      className="plus"
      onClick={[Function]}
    >
      +
    </button>
    <button
      className="equals"
      onClick={[Function]}
    >
      =
    </button>
  </div>
</div>
`;
